types/mongodb/index.d.ts

```diff
@@ -1312,7 +1313,7 @@ export type UpdateQuery<TSchema> = {
     $rename?: { [key: string]: string };
     $set?: MatchKeysAndValues<TSchema>;
     $setOnInsert?: MatchKeysAndValues<TSchema>;
-    $unset?: OnlyFieldsOfType<TSchema, any, ''>;
+    $unset?: OnlyFieldsOfType<TSchema, any, any>;
```

Comment:
Hmmm, I know that technically you can pass in anything, but I would personally prefer to be a bit more strict here.

e.g. if an instance of a class with `valueOf` or `toJSON` returning `undefined` is sent in, will the value be removed? Will `update` throw an error if a circular referencing object is passed in?

How would you feel about `'' | 1 | true` which seems to be the only usages I've seen in the wild?

That way people can't be confused by passing `false` and still having the field removed.